### ISSUE TITLE
[BUG] before_event of type Update

### ISSUE THREAD

**Issue by RempelOliveira** [BeanieODM/beanie#913]
**Describe the bug**

When using "before event" of type "Updated", the values ​​of the fields in the "self" variable aren't the values ​​passed in the request, they are the values ​​readed from the database.

**To Reproduce**

```
Database Records:
[{
   "_id": "123",
    "name": "Name 01"
}]
```
```
db_record = await Model.find_one(Model.id == "123")
await db_record.set({"name": "Name 02"})

class Model(Document):
    name: str

    @before_event(Update)
    def check_activation(self) -> "Model":
        return self  # self.name will be the value "Name 01" instead of "Name 02"
```

**Expected behavior**

My understanding is that this event serves to manipulate the input data before being inserted into the database, however, only when using the "Replace" type is it possible to have access to them, but I wouldn't like to use replace because I just need to update a few fields, not all the collection.

**Additional context**

My understanding is that this event serves to manipulate the input data before being inserted into the database, however, only when using the "Replace" type is it possible to have access to them, but I wouldn't like to use replace because I just need to update a few fields, not all the collection.

Using this event with "Insert" I can also access the request data, I can't only using "Update" ;/

Am I wrong about how I see this feature or is it a bug?
If it is the expected behavior, how can I do something like the example provided?

**Comment by RempelOliveira** [comment]
I just realized that this event can't save changes as well.

**Comment by roman-right** [comment]
Hi @RempelOliveira ,
This event can be used for some other things, like calling api before update and something like this. It can not change the update query itself and doesn't know the state of the object after query be applied

**Comment by gegnew** [comment]
🤯 I just also ran into this issue. I'm fairly sure this worked on a previous version; did something change?

**Comment by RempelOliveira** [comment]
So this odm is very good, but if the maintainer doesn't pay attention to its users' reports, it will become obsolete very quickly, there are some basic things to improve and fix and honestly, I'm looking for another odm because of this.

**Comment by dakohhh** [comment]
@RempelOliveira , if you ever find a soliutio, please reply to this comment, you'll be of great help 😢🙏🏼

**Comment by dakohhh** [comment]
@roman-right , is there an alternative way to go about this, such as overiding the update or save function ?

**Comment by RempelOliveira** [comment]
@dakohhh I can't remember exactly what I did, but I looked at the code now and I know that I still can't code this way because it hasn't been fixed.

For me it's a bug, but the package owner disagrees.

Anyway, in my project I had a repository layer where I had all the database operations and to update a record, I did it inside a method implemented in that layer.

For example, to update a data group and a datetime field, I just merge the data received as a parameter with the field value, something like the code below...

```
    async def update(self, data: AuthorizationDto.Update, parameters: AuthorizationDto.ReadOne) -> Authorization:
        authorization = await self.read_one(parameters)

        return await authorization.set({
            **data.model_dump(exclude_unset=True), Authorization.updated_at: DateTime.now()}, session=self.db_session
        )
```

In other words, I have to handle the data before updating the record and outside of the model.

If you want something a little more complex, you can create a decorator to intercept the data before executing the repository method, but it will depend on your needs.

**Comment by DominikDiebolder** [comment]
Still an open issue for the Update Event which is not persisting the mutations in the @before_event decorator.
